src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const StateCopy = { ...state };
+  const res = [];
+
+  for (const key of actions) {
+    if (key.type === 'addProperties') {
+      for (const i in key.extraData) {
```

Comment:
Using `for...in` to iterate over `key.extraData` is not ideal if `extraData` is an array. Consider using `for...of` or `Object.keys()` to iterate over the properties if `extraData` is an object.